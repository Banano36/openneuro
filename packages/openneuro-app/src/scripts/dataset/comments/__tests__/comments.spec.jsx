import React from 'react'
import { render } from '@testing-library/react'
import Comments from '../comments.jsx'

const emptyState =
  '{"blocks":[{"key":"3sm42","text":"","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}'
const exampleComment = {
  id: 'xyz',
  text: emptyState,
  user: { id: '1234', email: 'example@example.com' },
  replies: [],
}

describe('Comments component', () => {
  it('renders one top level comment', () => {
    const { asFragment } = render(
      <Comments datasetId="ds000001" comments={[exampleComment]} />,
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it('recursively renders a tree of comments', () => {
    const comments = [{ ...exampleComment, replies: [exampleComment] }]
    const { asFragment } = render(
      <Comments datasetId="ds000001" comments={comments} />,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
