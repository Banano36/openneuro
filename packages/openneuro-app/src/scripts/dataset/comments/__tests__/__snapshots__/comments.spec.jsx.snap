// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Comments component > recursively renders a tree of comments 1`] = `
<DocumentFragment>
  <div
    class="dataset-comments"
  >
    <h2>
      Comments
    </h2>
    <div>
      Please sign in to contribute to the discussion.
    </div>
  </div>
</DocumentFragment>
`;

exports[`Comments component > renders one top level comment 1`] = `
<DocumentFragment>
  <div
    class="dataset-comments"
  >
    <h2>
      Comments
    </h2>
    <div>
      Please sign in to contribute to the discussion.
    </div>
  </div>
</DocumentFragment>
`;
